skimage/graph/_mcp.pyx

```diff
@@ -586,8 +592,9 @@ cdef class MCP:
                 cost = flat_costs[index]
                 new_cost = flat_costs[new_index]
 
-                # If the cost at this point is negative or infinite, ignore it
-                if new_cost < 0 or new_cost == inf:
+                # If the cost at this point is negative, infinite or above max_cost, ignore it
+                above_max_cost = False if max_cost is None else new_cost > max_cost
+                if new_cost < 0 or new_cost == inf or above_max_cost:
                     continue
```

Comment:
I'd expect `max_cost=0` to filter out everything even if this seems like an edge case. So the check should probably differentiate between 0 and `None`: `max_cost is not None`.

Also, the logic might be simpler to follow by leaving the original if condition and just adding a new check if `max_cost` is given.

```suggestion
                if new_cost < 0 or new_cost == inf:
                    continue  # Ignore negative and inf cost
                if max_cost is not None and new_cost > max_cost:
                    continue  # Ignore, if it exceeds `max_cost`
```
Thoughts?

Reply:
> I'd expect max_cost=0 to filter out everything even if this seems like an edge case. So the check should probably differentiate between 0 and None: max_cost is not None.

`above_max_cost` does check whether `max_cost` is `None`, right? Or do you mean something else?

Reply:
It is fine with me to improve the readability. I replaced the ternary expression.